src/test/components/panels/EventOverviews.test.tsx

```diff
@@ -0,0 +1,217 @@
+}) => {
+  switch (type) {
+    case "Past":
+      return [
+        "Manage",
+        `${event.title} - ${event.subtitle}`,
+        "Tue, 12 Sept 2023, 16:30",
+        "Mon, 11 Sept 2023, 16:30",
+        event.location?.address?.town,
+        event.eventStatus,
+        `${event.numberOfConfirmedBookings} / ${event.numberOfPlaces}`,
+        event.numberOfWaitingListBookings,
+        event.numberAttended,
+        event.numberAbsent,
+        `${
+          (event.numberAttended / (event.numberAttended + event.numberAbsent)) *
+          100
+        }%`,
+      ];
+    case "Future":
+    default:
+      return [
+        "Manage",
+        `${event.title} - ${event.subtitle}`,
+        "Thu, 9 Sept 9999, 20:00",
+        "Thu, 9 Sept 9999, 20:00",
+        event.location?.address?.town,
+        event.eventStatus,
+        `${event.numberOfConfirmedBookings} / ${event.numberOfPlaces}`,
+        event.numberOfWaitingListBookings,
+        event.numberAttended,
+        event.numberAbsent,
+        "-",
+      ];
+  }
+};
+
+describe("Admin Event Manager", () => {
+  const mockSetSelectedEventId = jest.fn();
+
+  const setupTest = (role: TestUserRole) => {
+    renderTestEnvironment({
+      role: role,
+      PageComponent: EventOverviews,
+      componentProps: {
+        setSelectedEventId: mockSetSelectedEventId,
+        user: { ...mockUser, loggedIn: true, role: role },
+      },
+      initialRouteEntries: ["/admin/events"],
+      extraEndpoints: [
+        rest.get(API_PATH + "/events/overview", (req, res, ctx) => {
+          const filter = req.url.searchParams.get("filter");
+          if (filter === "FUTURE") {
+            return res(ctx.status(200), ctx.json(mockFutureEventOverviews));
+          } else if (filter === "PAST") {
+            return res(ctx.status(200), ctx.json(mockPastEventOverviews));
+          }
+        }),
+      ],
+    });
+  };
+
+  it("renders for Admin user without crashing", () => {
+    setupTest("ADMIN");
+    const subtitle = screen.getByText("Events overview");
+    expect(subtitle).toBeInTheDocument();
+  });
+
+  it("renders warning for Event Leader user", () => {
+    setupTest("EVENT_LEADER");
+    const eventLeaderWarning = screen.queryByText(
+      "As an event leader, you are only able to see the details of events which you manage."
+    );
+    expect(eventLeaderWarning).toBeInTheDocument();
+  });
+
+  it("displays table with correct headers", async () => {
+    setupTest("ADMIN");
+    await screen.findByText("Actions");
+    headers.forEach((header) => {
+      expect(
+        screen.getByRole("columnheader", {
+          name: header,
+        })
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("displays correct number of 'future' events", async () => {
+    setupTest("ADMIN");
+    await screen.findByText("Actions");
+    const table = screen.getByRole("table");
+    const rows = screen.getAllByRole("row", table);
+    expect(rows.length).toBe(2);
```

Comment:
Always use the more specific assertion, as they provide better error messages. In this case it would be:
```typescript
expect(rows).toHaveLength(2);
```

The are more tests in this file with this problem.